### DISCUSSION TITLE
Activate accesspoint only if no wifi connexion is found

### DISCUSSION THREAD

**Discussion by DaFrenchFrog** [RaspAP/raspap-webgui#1617] [Q&A]
Hi,

I am trying to allow a wifi setup on my Raspberry zero W. My intent is to install RaspAP and nodogsplash so when connecting to the AP you could enter your wifi settings and then reboot so it is connected to the "normal" wifi.

But so far I am stuck finding how to prevent RaspAP activation upon startup. I found a rasapd.service but even disabled it does not change. I also noticed there was several services installed (dnsmasq, hostapd,netfilter-persistent) but I'm afraid if I disable these services I will have no access to the rpi anymore...

I saw [this discussion ](https://github.com/RaspAP/raspap-webgui/discussions/1610) and I do not use a desktop version (Raspbian GNU/Linux 11 (bullseye) 6.1.21+)

Does anyone have any idea on how to disable raspap by default, and only activate the AP if there is no known wifi found ?

Thanks a lot for your help !

**Comment by billz** [comment]
There is no functionality to activate the AP when no known Wi-Fi networks are present. Best advice is to use a second wireless adapter.

**Comment by DaFrenchFrog** [comment]
Thank you for your answer.

I was expecting to make service that would check if there is a wifi connected (using `iwgetid`or alike), and if there is no wifi detected, launch raspAP service, let the user connect to raspi-webgui with a captive portal, the portal shows a page with inputs to enter wifi SSID and password, write them into wpa_supplicant, and reboot the rpi so it is eventually connected to internet.

Isn't it possible ?

**Reply by billz** [reply]
Yes it's possible to create this yourself. Again, this functionality is not part of RaspAP.

**Reply by DaFrenchFrog** [reply]
Sure, so my question is how do I disable RaspAP so my rpi will connect to wifi by default ? And how do I "manually" launch RaspAP so I can have a proper AP ?

**Reply by billz** [reply]
> my question is how do I disable RaspAP

```
sudo systemctl stop raspapd.service
sudo systemctl disable raspapd.service
```

> how do I "manually" launch RaspAP

```
sudo systemctl start raspapd.service
```

This assumes you accepted the default install option to add this service. If not, you may enable it manually (read the docs) or individually stop the services it manages.

**Reply by DaFrenchFrog** [reply]
That's what I tried and I disabled raspapd.service but upon reboot I still have the raspi-webgui network available and when I connect to the rpi with ssh and enter `systemctl status raspapd` it outputs :
```
raspapd.service - RaspAP Service Daemon
     Loaded: loaded (/lib/systemd/system/raspapd.service; disabled; vendor preset: enabled)
     Active: inactive (dead)
```
So the raspapd service is indeed disabled but the AP is still active I don't know why...

**Reply by billz** [reply]
Try `sudo systemctl disable hostapd`

**Reply by DaFrenchFrog** [reply]
Thanks a lot !

**Reply by DaFrenchFrog** [reply]
Sorry. I just figured out that now my application can't call external URLs. 

When I make a supabase request it throws an error :  `[Errno 113] No route to host`

Did the raspAP change some config appart from the services it runs ?

**Reply by billz** [reply]
RaspAP applies very basic [NAT rules](https://docs.raspap.com/manual/#routing-and-ip-masquerading), that's all. Review this plus the manual install steps. You may also want to check your `/etc/resolv.conf`. Contrary to popular belief, RaspAP only touches configs for the services it installs (hostapd, dhcp, dnsmasq).

**Reply by DaFrenchFrog** [reply]
OK it's a bit out of my league but I can search from here. What should I check in `/etc/resolv.conf` ?

**Reply by DaFrenchFrog** [reply]
I happen to have two rpis with the same starting config, one with RaspAP and the other one without, and I found that /etc/dhcpcd.conf seems to be modified by raspap.
Original file
````
hostname
clientid
persistent
option rapid_commit

option domain_name_servers, domain_name, domain_search, host_name
option classless_static_routes
option interface_mtu

require dhcp_server_identifier

slaac private
````
RaspAP modified file :
````
hostname
clientid
persistent
option rapid_commit
option domain_name_servers, domain_name, domain_search, host_name
option classless_static_routes
option ntp_servers
require dhcp_server_identifier
slaac private
nohook lookup-hostname

# RaspAP wlan0 configuration
interface wlan0
static ip_address=10.3.141.1/24
static routers=10.3.141.1
static domain_name_server=9.9.9.9 1.1.1.1
````

If I comment RaspAp lines and restart my app works normally, even if I activate raspapd service.
So it seems that these lines are not useful for my case. 
Do you confirm ?

**Reply by billz** [reply]
As already mentioned, dhcp is one of the services managed by RaspAP. I'd suggest spending some time to understand the [default settings](https://docs.raspap.com/defaults/). If your application is throwing a connection error, start with the most basic ping test and proceed from there.
